### ISSUE TITLE
convert_datatype changes number value to INTEGER.MAX when its too large

### ISSUE THREAD

**Issue by ppf2** [logstash-plugins/logstash-filter-dissect#12]
convert_datatype takes either int or float options today.

However, if you run a long value through the convert_datatype, it allows it.  However, the dissect -> convert_datatype will change the value of the field to INTEGER.MAX

Source value:

```
  "supportID" => "4394740425750718628"
```

Output value:

```
"supportID" => 2147483647
```

To workaround this, instead of using convert_datatype for dissect, I am post-processing it using mutate/convert:

```
  mutate {
    convert => { "supportID" => "integer" }
  }
```

Which returns the correct result:

```
"supportID" => 4394740425750718628
```

If we do not intend to support longs like mutate/convert does,  it will be better if we throw an error on the configuration instead of auto-converting the value to integer.max.

**Comment by guyboertje** [comment]
You are correct. Will fix.

**Comment by guyboertje** [comment]
Will be fixed in #34

**Comment by guyboertje** [comment]
fixed with #34
